src/panels/lovelace/cards/hui-entity-card.ts

```diff
@@ -0,0 +1,250 @@
+
+import { LovelaceCard, LovelaceCardEditor } from "../types";
+import { HomeAssistant } from "../../../types";
+import { fireEvent } from "../../../common/dom/fire_event";
+import { EntityCardConfig } from "./types";
+import { hasConfigOrEntityChanged } from "../common/has-changed";
+import { actionHandler } from "../common/directives/action-handler-directive";
+import { isValidEntityId } from "../../../common/entity/valid_entity_id";
+import { computeStateValue } from "../../../common/entity/compute_state_value";
+
+@customElement("hui-entity-card")
+class HuiEntityCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import(
+      /* webpackChunkName: "hui-entity-card-editor" */ "../editor/config-elements/hui-entity-card-editor"
+    );
+    return document.createElement("hui-entity-card-editor");
+  }
+
+  public static getStubConfig(): object {
+    return { entity: "" };
+  }
+
+  @property() public hass?: HomeAssistant;
+  @property() private _config?: EntityCardConfig;
+
+  public setConfig(config: EntityCardConfig): void {
+    if (config.entity && !isValidEntityId(config.entity)) {
+      throw new Error("Invalid Entity");
+    }
+
+    this._config = config;
+  }
+
+  public getCardSize(): number {
+    return 2;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.entity_not_found",
+            "entity",
+            this._config.entity
+          )}</hui-warning
+        >
+      `;
+    }
+
+    if (
+      this._config.attribute &&
+      !stateObj.attributes[this._config.attribute] &&
+      !this._config.default
+    ) {
+      return html`
+        <hui-warning
+          >${this.hass.localize(
+            "ui.panel.lovelace.warning.attribute_not_found",
+            "attribute",
+            this._config.attribute
+          )}</hui-warning
+        >
+      `;
+    }
+
+    const attributeValue = this._config.attribute
+      ? stateObj.attributes[this._config.attribute] || this._config.default
+      : this._config.default;
+
+    return html`
+      <ha-card
+        @action=${this._handleClick}
+        .actionHandler=${actionHandler()}
+        tabindex="0"
+      >
+        <div class="flex header">
+          <div class="name">
+            <span>${this._config.name || computeStateName(stateObj)}</span>
+          </div>
+          <div class="icon">
+            <ha-icon
+              .icon="${this._config.icon || stateIcon(stateObj)}"
+            ></ha-icon>
+          </div>
+        </div>
+        <div class="flex info">
+          <span id="value"
+            >${this._config.attribute
+              ? attributeValue
+              : computeStateValue(
```

Comment:
The downside of this approach imo is that we put a lot of work on the users, instead of us doing the right thing. This method is a step in the right direction, but it's still not aggressive enough. We can do better.

Let's start with creating pre-made configurations.

You pick a climate entity: offer the user a dropdown if they want to see the HVAC Mode, action, current temperature, current humidity (if available) or target temp.

Then there is also a custom mode where it's all whatever. Let's start with just the custom mode in this PR: give us an attribute or we just render the state. No data processing there. If it's state we're rendering, we can default to unit to the unit_of_measurement attribute, otherwise empty.

Reply:
So you are saying to remove any processing to make the data look nice. Ex. `off` -> `Off`
And remove the ability to give a Unit of measurement

Reply:
Remove all the magic.